### PR TITLE
Increment wallet key manager index during recovery

### PR DESCRIPTION
**Pull request by philipr-za**
## Description
Currently, when recovery is performed the key manager index is not incremented as outputs are found. This means that when new transactions are performed the key manager produces previously used spending keys for the new outputs, reusing private keys. We also did not apply the correct script private key to the recovered output. This wasn’t a problem because we are using `Nop` scripts but for any script that requires this key to be correct we would have had a problem.

This PR adds the logic that when an output is rewound during recovery that we find the key manager index for the recovered spending key, update the key manager index to one more than the highest index found during recovery and also use this index to apply the correct script private key to the recovered output.

This PR also contains a but of neatness refactoring in the Output Manager Service.

## How Has This Been Tested?
Existing tests

## Checklist:

* [ ] I'm merging against the `development` branch.
* [x] I have squashed my commits into a single commit.

### CODE REVIEW

**Review by delta1 — CHANGES_REQUESTED** [review]
Looks great! The update to `package.json` means that you'll have to commit `package-lock.json` as part of this change, `npm ci` installs directly from the lockfile which is out of date now

**Comment by philipr-za** [comment]
> Looks great! The update to `package.json` means that you'll have to commit `package-lock.json` as part of this change, `npm ci` installs directly from the lockfile which is out of date now

Done

**Review by delta1 — APPROVED** [review]
LGTM

File: base_layer/wallet/src/output_manager_service/master_key_manager.rs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        let mut skm = self.coinbase_script_key_manager.lock().await;
+        let script_key = skm.next_key()?;
+        Ok((spending_key.k, script_key.k))
+    }
+
+    /// Return the Seed words for the current Master Key set in the Key Manager
+    pub async fn get_seed_words(&self) -> Result<Vec<String>, OutputManagerError> {
+        Ok(from_secret_key(
+            self.utxo_key_manager.lock().await.master_key(),
+            &MnemonicLanguage::English,
```
Comment (SWvheerden) [inline base_layer/wallet/src/output_manager_service/master_key_manager.rs]: nit: Should this not rather be a config option for `MasterKeyManager`. 
This seems like a smell, hard-coded option within a function. 
I would rather say let `MasterKeyManager` have a field language and hard code it in there for now.

File: base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        Self {
+            master_key_manager,
+            factories,
+            db,
+        }
+    }
+
+    /// Attempt to rewind all of the given transaction outputs into unblinded outputs. If they can be rewound then add
+    /// them to the database and increment the key manager index
+    pub async fn scan_and_recover_outputs(
```
Comment (SWvheerden) [inline base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs:70]:
```suggestion
    pub async fn scan_recover_and_add_outputs(
```

This function now scans, recovers, and adds the outputs it finds. 
I am not completely for this as this function now does more than one thing, but we should at least update the name to reflect that.

**Review by SWvheerden — COMMENTED** [review]
Looks good, 2 nits thou.

File: base_layer/wallet/src/output_manager_service/master_key_manager.rs (hunk 1 above)
Comment (philipr-za) [reply base_layer/wallet/src/output_manager_service/master_key_manager.rs]: Fair enough, gave this function a argument to specify the language and added it to the Output Manager Config and its a default now.

File: base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs (hunk 2 above)
Comment (philipr-za) [reply base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs:70]: I will argue that the verb `recover` is the process of adding it to the database. The method does to two things, `scans` which is going through and trying to rewind an output and `recover` which is adding it to the db and updating the keymanager. I thought about splitting the second part out but decided that I couldn't think of a use case for every allowing these to happen in isolation and that the interface should only allow them to be done together.

Comment (delta1) [reply base_layer/wallet/src/output_manager_service/recovery/standard_outputs_recoverer.rs:70]: I agree with @philipr-za , "recover" here already means "recover for spending" which means add them to your wallet.

File: base_layer/wallet/src/output_manager_service/handle.rs [hunk 3]
```diff
@@ -103,7 +103,7 @@ impl fmt::Display for OutputManagerRequest {
             GetCoinbaseTransaction(_) => write!(f, "GetCoinbaseTransaction"),
             GetPublicRewindKeys => write!(f, "GetPublicRewindKeys"),
             FeeEstimate(_) => write!(f, "FeeEstimate"),
-            RewindOutputs(_, _) => write!(f, "RewindAndImportOutputs"),
+            ScanForRecoverableOutputs(_, _) => write!(f, "ScanAndImportRecoverableOutputs"),
```
Comment (delta1) [inline base_layer/wallet/src/output_manager_service/handle.rs]: just noticed these don't match, not sure it's an issue though

Comment (philipr-za) [reply base_layer/wallet/src/output_manager_service/handle.rs]: fixed

**Pull request merged**